CHANGELOG.md

```diff
@@ -12,6 +12,13 @@
   `DeviceSubType::MESSAGING_NVME_OF_NAMESPACE`.
 - Added `MemoryAttribute::SPECIAL_PURPOSE`, `MemoryAttribute::CPU_CRYPTO`,
   `MemoryAttribute::ISA_VALID`, and `MemoryAttribute::ISA_MASK`.
+- Added `BootServices::install_protocol_interface`,
+  `BootServices::uninstall_protocol_interface`, and
+  `BootServices::reinstall_protocol_interface`.
+- Added `BootServices::register_protocol_notify`.
+- Added `SearchType::ByRegisterNotify` and 
+  `SearchType::from_search_key`.
+- Added `SearchKey` type.
```

Comment:
Nit: I think this last line can be dropped since it's sort of a detail related to `from_search_key`.

Reply:
It is part of the public API because of `register_protocol_notify` though. I could return a `SearchType::ByRegisterNotify` instead and make `ProtocolSearchKey` private. That might be better for the user too, as any time they need to search by key they would have to turn it into a `SearchType` anyway.

Reply:
I think having the public type is fine, I just mean it's probably not worth mentioning in the changelog since the previous line about `SearchType::ByRegisterNotify` is really the important part. Maybe just combine the SearchType and SearchKey lines to make clear they are connected.